sklearn/linear_model/ransac.py

```diff
@@ -243,6 +247,17 @@ def fit(self, X, y):
         except ValueError:
             pass
 
+        estimator_fit_has_sample_weight = has_fit_parameter(base_estimator, "sample_weight")
+        estimator_name = type(base_estimator).__name__
+        if (sample_weight is not None
+                    and not estimator_fit_has_sample_weight ):
+            raise ValueError("%s does not support sample_weight. Samples"
+                              " weights are only used for the calibration"
+                              " itself." % estimator_name)
+            base_estimator_sample_weight = None
+        else:
```

Comment:
can we do away with `base_estimator_sample_weight`? we should also support `sample_weight` of non-numpy array types.

    if sample_weight is not None:
        sample_weight = np.asarray(sample_weight)